### PR TITLE
PR for Windows automation 

### PR DESCRIPTION
**Pull request by mdrahman-suse**
## Problem

Currently there is no support to add Windows node as agent to a rke2 standalone cluster. The process of adding the agent is tedious and time consuming, specially during rke2 patch or minor release validation.
 
## Solution

This PR adds the ability to automatically add Windows node as agent to a rke2 cluster

### CODE REVIEW

**Review by rancher-max — CHANGES_REQUESTED** [review]

**Review by VestigeJ — COMMENTED** [review]
LGTM after those two tweaks to change the url and remove sleep from the equation.

File: tests/validation/tests/v3_api/resource/terraform/rke2/windows_worker/join-rke2.tftpl
Comment (rancher-max) [inline tests/validation/tests/v3_api/resource/terraform/rke2/windows_worker/join-rke2.tftpl:8]: Let's get this PRed and merged into `rancher/rke2` before merging this PR so that we can pull from there instead of your fork.

File: tests/validation/tests/v3_api/test_import_rke2_cluster.py
Comment (rancher-max) [inline tests/validation/tests/v3_api/test_import_rke2_cluster.py:1]: Let's remove this since we aren't using it anymore

**Review by rancher-max — CHANGES_REQUESTED** [review]

File: tests/validation/tests/v3_api/test_import_rke2_cluster.py
Comment (mdrahman-suse) [reply tests/validation/tests/v3_api/test_import_rke2_cluster.py]: Sure, removed 💯

**Comment by mdrahman-suse** [comment]
@rancher-max @ShylajaDevadiga please review, made some changes to retry logic

File: tests/validation/tests/v3_api/resource/terraform/rke2/windows_worker/join-rke2.tftpl
Comment (rancher-max) [reply tests/validation/tests/v3_api/resource/terraform/rke2/windows_worker/join-rke2.tftpl:8]: We still need to do this

File: tests/validation/tests/v3_api/test_import_rke2_cluster.py
Comment (rancher-max) [inline tests/validation/tests/v3_api/test_import_rke2_cluster.py:275]: In your testing, were there any flakes based on the timings? I feel like we should also wait for 5 minutes on the node/pod Ready/Running statuses too instead of just 2 minutes there since RKE2 can be slow

Comment (mdrahman-suse) [reply tests/validation/tests/v3_api/test_import_rke2_cluster.py:275]: I have not seen any flakes based on timing. But I will take your suggestion to add 5 minutes to nodes and pods checking to be safe... totally agree on rke2 slowness :p

**Review by rancher-max — DISMISSED** [review]

**Review by rancher-max — APPROVED** [review]
LGTM

**Pull request merged**
